packages/utils/src/signatureUtils.ts

```diff
@@ -1,6 +1,55 @@
 const ethjsutil = require('ethereumjs-util')
 
-export function signMessage(messageHash: string, privateKey: string, address: string) {
+import * as Web3Utils from 'web3-utils'
+import { privateKeyToAddress } from './address'
+
+// If messages is a hex, the length of it should be the number of bytes
+function messageLength(message: string) {
+  if (Web3Utils.isHexStrict(message)) {
+    return (message.length - 2) / 2
+  }
+  return message.length
+}
+// Ethereum has a special signature format that requires a prefix
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign
+export function hashMessageWithPrefix(message: string) {
+  const prefix = '\x19Ethereum Signed Message:\n' + messageLength(message)
+  return Web3Utils.soliditySha3(prefix, message)
+}
+
+export function hashMessage(message: string): string {
+  return Web3Utils.soliditySha3({ type: 'string', value: message })
+}
+
+export interface Signer {
+  sign: (message: string) => Promise<string>
+}
+
+// Uses a native function to sign (as signFn), most commonly `web.eth.sign`
+export function NativeSigner(
+  signFn: (message: string, signer: string) => Promise<string>,
+  signer: string
+): Signer {
+  return {
+    sign: async (message: string) => {
+      return signFn(message, signer)
+    },
+  }
+}
+export function LocalSigner(privateKey: string): Signer {
+  return {
+    sign: async (message: string) =>
+      Promise.resolve(
+        serializeSignature(signMessage(message, privateKey, privateKeyToAddress(privateKey)))
+      ),
+  }
+}
+
+export function signMessage(message: string, privateKey: string, address: string) {
+  return signMessageNoPrefix(hashMessageWithPrefix(message), privateKey, address)
```

Comment:
Hmm maybe my previous suggestion was misguided because this line is really confusing:
```signMessageNoPrefix(hashMessageWithPrefix```